src/compiler/checker.ts

```diff
@@ -3862,6 +3925,23 @@ namespace ts {
         }
 
         function isOptionalParameter(node: ParameterDeclaration) {
+            if (node.parserContextFlags & ParserContextFlags.JavaScriptFile) {

```

Comment:
I never liked the extra `parserContextFlags` property that Cyrus introduced. Those flags should just be in the regular `flags` property where there is ample room. The problem is that most nodes don't have the property, so this will be a cache miss most of the time when you're not a JavaScript file. We really should just consolidate the two.